### DISCUSSION TITLE
Is there a way to disable horizontal swipe/gesture for cities?

### DISCUSSION THREAD

**Discussion by KishanPRao** [breezy-weather/breezy-weather#1035] [Q&A]
I'd like to know if there's a way to disable the horizontal gesture to change cities?

I find it sometimes conflicting with other horizontal gestures (like daily forecast).
This is noticeable if you fling (short, quick swipe) horizontally.

I'd prefer to simply disable it instead.

**Comment by papjul** [comment]
It was made so that there is no risk of swiping to the next location if you are swiping inside a horizontal scrollable area, let us know if that's not the case

**Reply by KishanPRao** [reply]
I was mistaken, it's not a short swipe, but only a fast swipe.
It's difficult to perfectly reproduce, but if I quick swipe left or right (irrelevant of if it's the end of the list of items), it sometimes triggers change of location.

**Reply by papjul** [reply]
Please record a video with visible touch inputs, thanks!

**Reply by KishanPRao** [reply]
https://streamable.com/j57nmf
It should be noticeable at 6 seconds.

**Reply by papjul** [reply]
I don't see any change of location

**Reply by min7-i** [reply]
I think @KishanPRao means that a swipe to switch a location is indicated - the animation where the page slightly moves left/right and then jumps back to its original position.

This is happening in two cases:

1. When the chart is at its starting or end position - which means you cannot scroll further to the left or right within the chart. In that case a swipe will trigger a location change.
2. If you swipe diagonally and the vertical distance travelled is greater than the horizontal distance, you'll either trigger a scroll (up/down) or a location change. I didn't check this again, but if I remember correctly, [this ](https://github.com/breezy-weather/breezy-weather/blob/26aff7e483fb33187a51b1b74000a886b1fdbb18/app/src/main/java/org/breezyweather/common/ui/widgets/SwipeSwitchLayout.kt#L113) should be the condition responsible for this behavior.

If you set the recording to 0.5x speed, you can see the second case [here](https://streamable.com/j57nmf?t=12) for instance.

**Reply by papjul** [reply]
> I think @KishanPRao means that a swipe to switch a location is indicated - the animation where the page slightly moves left/right and then jumps back to its original position.

We are not talking about the bouncing effect, which is normal when you swipe outside the horizontal area and is designed to prevent any accidental swipe to next location, but:

> it sometimes triggers change of location

This is what I want to see in video

**Reply by KishanPRao** [reply]
@min7-i thank you for clarifying my issue, and breaking down what occurs!
@papjul I do also mean the bouncing effect shouldn't occur, as swiping the inner item shouldn't technically affect the outer tab/location area.

> This is what I want to see in video

[This](https://streamable.com/aebaje?t=30) highlights the scenario where it changes (might need viewing frame-by-frame).
This might be an edge case, where it actually changes to a different location while swiping an inner view.

[Here](https://streamable.com/aebaje?t=6), it changes location when it's already at the end position.

The bigger issue for me is [this](https://streamable.com/j57nmf?t=6), where you swipe right-to-left, and it causes a bounce effect.
This shouldn't occur because there are still items remaining on the right, and I'm swiping within the horizontal area.

**Reply by papjul** [reply]
Alright, but if I understand correctly:
- It never swipes to next location
- There is no boucing effect when you swipe normally
- There is only a minor bouncing effect that is triggered when you swipe very fast

Some people requested the opposite of what you request: being able to swipe to to next location when doing a very fast swipe, while keeping horizontal scroll for normal swipes. And I agree with them, this is how many launchers work (they require a shorter distance if the swipe was fast), so I would recommend you go easy on the swipe 😅

Unfortunately, Geometric Weather was designed with two horizontal scroll events; it's not the best in terms of UX, so maybe at some point, we will have to replace them with 24-hourly chart (already planned) and vertical daily charts like most weather apps

**Reply by KishanPRao** [reply]
> It never swipes to next location

[Here](https://streamable.com/aebaje?t=30), it swipes to next location.

> There is no boucing effect when you swipe normally

I guess I normally swipe slightly faster. :smile: 

I do think swiping is a better approach than tapping to change location. But swiping/bouncing of the outer layout when the inner layout is interacted feels off.

I tried to add a setting to disable the swipe [here](https://github.com/KishanPRao/breezy-weather/tree/disable-location-gesture).
I can open a PR, if this isn't antithetical to the app's UX.

I'd like to try and fix the touch events, so that one doesn't affect the other, but I'm slightly rusty with android UI events.
And, even with the disabled approach, I can see that slightly diagonal swipes cause vertical swipes.

**Reply by papjul** [reply]
> [Here](https://streamable.com/aebaje?t=30), it swipes to next location.

It's so fast that I don't even understand what you're doing ^^' I don't think that's a normal use case

> I'd like to try and fix the touch events, so that one doesn't affect the other,

You mean implementing this?

> being able to swipe to to next location when doing a very fast swipe, while keeping horizontal scroll for normal swipes. And I agree with them, this is how many launchers work (they require a shorter distance if the swipe was fast)

> I tried to add a setting to disable the swipe [here](https://github.com/KishanPRao/breezy-weather/tree/disable-location-gesture).

We don't add this kind of preferences. My approach is to remove horizontal scrollable cards:

> replace them with 24-hourly chart (already planned) and vertical daily charts like most weather apps

**Reply by KishanPRao** [reply]
> It's so fast that I don't even understand what you're doing

Haha, sorry, I was trying to reproduce it, as it normally occurred randomly.

> My approach is to remove horizontal scrollable cards

I think that'd be the best solution, I'm looking forward to it!

**Reply by papjul** [reply]
Tracked as #1037
